plugin/security/src/main/java/org/opencds/cqf/ruler/security/interceptor/RulerExceptionHandlingInterceptor.java

```diff
@@ -0,0 +1,126 @@
+package org.opencds.cqf.ruler.security.interceptor;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.context.FhirVersionEnum;
+import ca.uhn.fhir.interceptor.api.Hook;
+import ca.uhn.fhir.interceptor.api.Interceptor;
+import ca.uhn.fhir.interceptor.api.Pointcut;
+import ca.uhn.fhir.jpa.api.dao.DaoRegistry;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.server.exceptions.BaseServerResponseException;
+import org.apache.commons.lang3.StringUtils;
+import ca.uhn.fhir.rest.server.exceptions.AuthenticationException;
+import org.hl7.fhir.instance.model.api.IBaseOperationOutcome;
+import org.opencds.cqf.ruler.behavior.DaoRegistryUser;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+
+@Interceptor
+public class RulerExceptionHandlingInterceptor implements org.opencds.cqf.ruler.api.Interceptor, DaoRegistryUser {
+
+	@Autowired
+	private DaoRegistry myDaoRegistry;
+
+	@Override
+	public DaoRegistry getDaoRegistry() {
+		return myDaoRegistry;
+	}
+
+	@Override
+	public FhirContext getFhirContext() {
+		return DaoRegistryUser.super.getFhirContext();
+	}
+
+	@Hook(Pointcut.SERVER_HANDLE_EXCEPTION)
+	public boolean handleException(RequestDetails theRequestDetails, BaseServerResponseException theException,
+											 HttpServletRequest theServletRequest, HttpServletResponse theServletResponse) throws IOException {
+
+		IBaseOperationOutcome operationOutcome = theException.getOperationOutcome();
+		if(operationOutcome == null || theException instanceof AuthenticationException) {
+			updateResponse(theServletResponse, theException, generateOperationOutcome(theException));
+		} else {
+			Throwable causedBy = getCause(theException);
+			String actualCause = causedBy.getMessage();
```

Comment:
I there a way to get the ID of the specific resource that's missing?

Reply:
Not found in the current stackTrace. It is possible to fetch ID by editing CqlException definition. Currently CqlException does not store the cause it just make a description out of it here: 
https://github.com/DBCG/cql_engine/blob/1832f12f0eb6f4146d50382837f6651f7eca1d86/engine/src/main/java/org/opencds/cqf/cql/engine/exception/CqlException.java#L16